repos/fdbt-netex-output/src/netex-convertor/netexGenerator.ts

```diff
@@ -370,12 +378,22 @@ const netexGenerator = async (ticket: Ticket, operatorData: Operator[]): Promise
         tariff.fareStructureElements.FareStructureElement = fareStructuresElements;
 
         if (isFlatFareType(ticket)) {
+            const isPricedByDistance = 'pricingByDistance' in ticket.products[0];
+            const typeOfTariffRef = isPricedByDistance ? 'fxc:Distance_kilometers' : 'fxc:flat';
+            const tariffBasis = isPricedByDistance ? 'distance' : 'flat';
+
             tariff.TypeOfTariffRef = {
                 version: 'fxc:v1.0',
-                ref: 'fxc:flat',
+                ref: typeOfTariffRef,
             };
 
-            tariff.TariffBasis = { $t: 'flat' };
+            tariff.TariffBasis = { $t: tariffBasis };
+
+            if (isPricedByDistance) {
+                tariff.geographicalIntervals.GeographicalInterval = getGeographicalIntervals(
+                    ticket.products[0] as PriceByDistanceProduct,
```

Comment:
can we move this up for clarity about what we are doing with Geographical unit in pricedbydistance